### PR TITLE
Improve footer

### PR DESCRIPTION
**Pull request by blacksponge**
## Purpose

Fix alignment issue and improve accessibility.

## Context

\-

## Changes

* Move footer links into a `<ul>` into a `<nav>`
* Add a link for documentation (was blocking https://github.com/zonemaster/zonemaster-gui/pull/383 and might be useful now that we have all public documentation in the same place)
* Fix alignment issue of the "Github" link and bad wrapping of the contact email on medium-small screen
* Use a  `<dl>` for program versions
* Improve color contrast for copyright text

Before
Screenshot 2023-04-25 at 12-44-47 Zonemaster

After
Screenshot 2023-04-25 at 12-45-05 toto com · Zonemaster

## How to test this PR

* Try to resize the viewport and check that the footer content is wrapping correctly.

### CODE REVIEW

File: src/app/components/footer/footer.component.html [hunk 1]
```diff
@@ -1,36 +1,38 @@
...
+        <li class="col-md">
+            <span placement="top" [ngbTooltip]="tipContent" i18n tabindex="0">Program versions</span>
+        </li>
+        <li class="col-md-auto">
+          <i class="fa fa-envelope-o" aria-hidden="true"></i>
+          <a class="text-dark" href="mailto:{{contactAddress}}">{{contactAddress}}</a>
+        </li>
+        <li class="col-md-auto">
+          <i class="fa fa-code" aria-hidden="true"></i>
+          <a class="text-dark" href="https://github.com/zonemaster/zonemaster">Source code</a>
```
[the remark this answers is not in the document]
Comment (blacksponge) [reply src/app/components/footer/footer.component.html:14]: Maybe, I don't have much opinion on this, anyone else?

[the remark this answers is not in the document]
Comment (matsduf) [reply src/app/components/footer/footer.component.html:14]: I think it is better to have the link to point to a generic place. In the README there is a link to the documentation. Instead if the link has to be updated we will have problem with old versions of GUI that point to old places.

**Comment by matsduf** [comment]
I am not fond of the program versions being displayed as tooltip. Besides that it is quite obscure to get the versions, it is also quite hard to copy the information. Could we let it fold out downwards instead?

File: src/app/components/footer/footer.component.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (blacksponge) [reply src/app/components/footer/footer.component.html:14]: (If there had been a README file in the `doc/public` I would have linked directly there, maybe there should be one?)

**Comment by matsduf** [comment]
> Honestly I don't like the tooltip either, in general they make discovery harder. I would prefer if the information would always be visible. Would you be okay with that? With a column with the software versions, and a second one with the links?

The software version is not interesting for the users in the normal case so it should not take too much room. I think a fold down similar to how e.g. "Zone" is folded out in a result, would be good. But I am not against having it always visible, but then I guess in quite small size.

**Comment by blacksponge** [comment]
@matsduf @tgreenx as we decided to not change the version tooltip in this PR in the last meeting can you re-review / approve ?

**Comment by pnax** [comment]
### v2023.1 release testing

Works nicely. A small glitch (that wasn't introduced with this PR, can be reproduced on master), the version tooltip goes off-screen on small width screen.

zm-gui-footer

**Pull request merged**